.pre-commit-config.yaml

```diff
@@ -3,12 +3,23 @@ repos:
   rev: v8.16.3
   hooks:
   - id: gitleaks
-- repo: https://github.com/shellcheck-py/shellcheck-py
-  rev: v0.10.0.1
+
+- repo: local
```

Comment:
I prefer using pre-commit and pre-commit-ci over GHA. I'm fine with adding things like typos (seeing how you fixed quite a few typos, it's a good idea), but please not running in GHA.